src/html5_playback.test.js

```diff
@@ -116,8 +134,26 @@ describe('HTML5TVsPlayback', function() {
     expect(Object.getOwnPropertyDescriptor(Object.getPrototypeOf(this.playback), 'duration').get).toBeTruthy()
   })
 
-  test('duration getter returns video.duration property', () => {
-    expect(this.playback.duration).toEqual(this.playback.el.duration)
+  describe('duration getter', () => {
+    test('returns video.duration property for VoD content', () => {
+      expect(this.playback.duration).toEqual(this.playback.el.duration)
+    })
+
+    test('returns the difference between video.seekable.end(0) and video.seekable.start(0) values for live content', () => {
```

Comment:
Just curious (I really don't know the reason), but why is the duration calculated as the length of the first seekable range but not as `end(last range) - start(first range)`?

Reply:
Yes, there can be more than one seekable range, but idk in which scenario it can happen

Reply:
Seems for the live content, the seekable ranges are merged into only one value during the streaming consumption. I don't know too why this happens but just to make the logic more clearer, I've applied this suggestion on this commit: 04f6b68